desktop/libs/hadoop/src/hadoop/fs/webhdfs.py

```diff
@@ -212,10 +214,34 @@ def current_trash_path(self, trash_path):
     return self.join(trash_path, self.TRASH_CURRENT)
 
   def _getparams(self):
-    return {
-      "user.name": WebHdfs.DEFAULT_USER,
-      "doas": self.user
-    }
+    if self._security_enabled:
+      try:
+        token = self.get_delegation_token(self.user)
+      except:
+        with self.get_delegation_token.cache_lock:
+          self.get_delegation_token.cache.pop(self.get_delegation_token.cache_key(self.user), None)
+      token = self.get_delegation_token(self.user)
+      return {
+        "delegation": token
+      }
+    
+    else:
+      return {
+        "user.name": WebHdfs.DEFAULT_USER,
+        "doas": self.user
+      }
+  
+  # a renew-interval (default is 24 hours)
+  # https://blog.cloudera.com/hadoop-delegation-tokens-explained/
+  @ttl_cache(maxsize=128, ttl=timedelta(hours=24), timer=datetime.now, typed=False)
+  def get_delegation_token(self, renewer):
```

Comment:
In order to get delegation token for the real user, `doas` param is also needed when security is enabled.
https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/WebHDFS.html#Proxy_Users